x/wasm/keeper/msg_server_integration_test.go

```diff
@@ -333,3 +333,106 @@ func TestMigrateContract(t *testing.T) {
 		})
 	}
 }
+
+func TestExecuteContract(t *testing.T) {
+	wasmApp := app.Setup(false)
+	ctx := wasmApp.BaseApp.NewContext(false, tmproto.Header{Time: time.Now()})
+
+	var (
+		myAddress       sdk.AccAddress = make([]byte, types.ContractAddrLen)
+		_, _, otherAddr                = testdata.KeyTestPubAddr()
+	)
+
+	specs := map[string]struct {
+		addr   string
+		expErr bool
+	}{
+		"adress can execute a contract": {
+			addr:   myAddress.String(),
+			expErr: false,
+		},
+		"other address cannot execute a contract": {
+			addr:   otherAddr.String(),
+			expErr: true,
+		},
+	}
+	for name, spec := range specs {
+		t.Run(name, func(t *testing.T) {
+			xCtx, _ := ctx.CacheContext()
+			// setup
+			_, _, sender := testdata.KeyTestPubAddr()
+			msg := types.MsgStoreCodeFixture(func(m *types.MsgStoreCode) {
+				m.WASMByteCode = hackatomContract
+				m.Sender = sender.String()
+			})
+
+			// store code
+			rsp, err := wasmApp.MsgServiceRouter().Handler(msg)(xCtx, msg)
+			require.NoError(t, err)
+			var storeCodeResponse types.MsgStoreCodeResponse
+			require.NoError(t, wasmApp.AppCodec().Unmarshal(rsp.Data, &storeCodeResponse))
+
+			// instantiate contract
+			initMsg := keeper.HackatomExampleInitMsg{
+				Verifier:    myAddress,
+				Beneficiary: otherAddr,
+			}
+			initMsgBz, err := json.Marshal(initMsg)
+			require.NoError(t, err)
+			msgInstantiate := &types.MsgInstantiateContract{
+				Sender: spec.addr,
+				Admin:  myAddress.String(),
+				CodeID: storeCodeResponse.CodeID,
+				Label:  "test",
+				Msg:    initMsgBz,
+				Funds:  sdk.Coins{},
+			}
+			rsp, err = wasmApp.MsgServiceRouter().Handler(msgInstantiate)(xCtx, msgInstantiate)
+			require.NoError(t, err)
+			var instantiateResponse types.MsgInstantiateContractResponse
+			require.NoError(t, wasmApp.AppCodec().Unmarshal(rsp.Data, &instantiateResponse))
```

Comment:
I think it is better to move this part out of the loop, because it is the test environment.